service_identity/src/main/kotlin/org/thechance/service_identity/api/endpoints/permissionRoutes.kt

```diff
@@ -0,0 +1,61 @@
+package org.thechance.service_identity.api.endpoints
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import org.koin.ktor.ext.inject
+import org.thechance.service_identity.api.model.PermissionDto
+import org.thechance.service_identity.domain.usecases.permission.PermissionUseCasesContainer
+
+fun Route.permissionRoutes() {
+    val permissionUseCasesContainer: PermissionUseCasesContainer by inject()
+    route("/permissions") {
+        post {
+            val permission = call.receive<PermissionDto>()
+            val success = permissionUseCasesContainer.addPermission(permission.toPermission())
+            if (success) {
+                call.respond(HttpStatusCode.Created)
+            } else {
+                call.respond(HttpStatusCode.InternalServerError)
+            }
+        }
+
+        delete("/permissions/{permissionId}") {
+            val permissionId = call.parameters["permissionId"] ?: return@delete call.respond(
+                HttpStatusCode.BadRequest
+            )
+            val success = permissionUseCasesContainer.deletePermission(permissionId)
+            if (success) {
+                call.respond(HttpStatusCode.NoContent)
+            } else {
+                call.respond(HttpStatusCode.InternalServerError)
+            }
+        }
+        put("/{permissionId}") {
+            val permissionId = call.parameters["permissionId"] ?: return@put call.respond(
+                HttpStatusCode.BadRequest
+            )
+            val permission = call.receive<PermissionDto>()
+            if (permission.id != permissionId) {
+                return@put call.respond(HttpStatusCode.BadRequest)
+            }
+            val success = permissionUseCasesContainer.updatePermission(
+                permissionId,
+                permission.toPermission()
+            )
+            if (success) {
+                call.respond(HttpStatusCode.NoContent)
+            } else {
+                call.respond(HttpStatusCode.InternalServerError)
+            }
+        }
+        get("/{id}") {
+            val permissionId = call.parameters["id"] ?: throw IllegalArgumentException("Invalid permission ID")
+            val permission = permissionUseCasesContainer.getPermission.invoke(permissionId).toPermissionDto()
```

Comment:
Is there a need for calling `invoke `here couldn't it just be normal brantheses invocation